packages/mobile/src/navigator/DrawerNavigator.tsx

```diff
@@ -0,0 +1,196 @@
+import fontStyles from '@celo/react-components/styles/fonts.v2'
+import {
+  createDrawerNavigator,
+  DrawerContentComponentProps,
+  DrawerContentOptions,
+  DrawerContentScrollView,
+  DrawerItemList,
+} from '@react-navigation/drawer'
+import BigNumber from 'bignumber.js'
+import * as React from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, Text, View } from 'react-native'
+import deviceInfoModule from 'react-native-device-info'
+import Account from 'src/account/Account'
+import FiatExchange from 'src/account/FiatExchange'
+import { default as InviteScreen } from 'src/account/Invite'
+import {
+  defaultCountryCodeSelector,
+  e164NumberSelector,
+  nameSelector,
+  userContactDetailsSelector,
+} from 'src/account/selectors'
+import Support from 'src/account/Support'
+import BackupIntroduction from 'src/backup/BackupIntroduction'
+import ExchangeHomeScreen from 'src/exchange/ExchangeHomeScreen'
+import WalletHome from 'src/home/WalletHome'
+import { Namespaces } from 'src/i18n'
+import AccountKey from 'src/icons/navigator/AccountKey'
+import AddWithdraw from 'src/icons/navigator/AddWithdraw'
+import Gold from 'src/icons/navigator/Gold'
+import Help from 'src/icons/navigator/Help'
+import Home from 'src/icons/navigator/Home'
+import Invite from 'src/icons/navigator/Invite'
+import Settings from 'src/icons/navigator/Settings'
+import { useDollarsToLocalAmount, useLocalCurrencySymbol } from 'src/localCurrency/hooks'
+import { Screens } from 'src/navigator/Screens'
+import useSelector from 'src/redux/useSelector'
+import { stableTokenBalanceSelector } from 'src/stableToken/reducer'
+import { currentAccountSelector } from 'src/web3/selectors'
+
+const Drawer = createDrawerNavigator()
+
+function CustomDrawerContent(props: DrawerContentComponentProps<DrawerContentOptions>) {
+  const displayName = useSelector(nameSelector)
+  const e164PhoneNumber = useSelector(e164NumberSelector)
+  const contactDetails = useSelector(userContactDetailsSelector)
+  const defaultCountryCode = useSelector(defaultCountryCodeSelector)
+  const balance = useSelector(stableTokenBalanceSelector)
+  const bigNumBalance = balance ? new BigNumber(balance) : undefined
+  const localBalance = useDollarsToLocalAmount(balance)
+  const symbol = useLocalCurrencySymbol()
+  const { t } = useTranslation(Namespaces.global)
+  const account = useSelector(currentAccountSelector)
+  const appVersion = deviceInfoModule.getVersion()
+
+  return (
+    <DrawerContentScrollView {...props}>
+      <View style={styles.drawerTop}>
+        <ContactCircle thumbnailPath={contactDetails.thumbnailPath} size={64} />
+        <Text style={[fontStyles.large600, styles.nameLabel]}>{displayName}</Text>
+        <PhoneNumberWithFlag
+          e164PhoneNumber={e164PhoneNumber}
+          defaultCountryCode={defaultCountryCode}
+        />
+        <View style={styles.border} />
+        <Text style={fontStyles.regular500}>{`${symbol} ${localBalance?.toFixed(2)}`}</Text>
```

Comment:
Can you use the existing money components for this instead of formatting it yourself. It'd be good to avoid the mess that @jeanregisser recently got us out of where money values are formatted 30 diff ways in diff places and don't handle local currency right.

Reply:
I looked at doing this using `CurrencyDisplay` but it seems like it adds a lot of complexity. @jeanregisser curious if you think it's necessary here - from my understanding we will always show both the celo dollars balance and the local amount, so i am not sure what edge cases would be possible (but maybe i am missing something)

Reply:
Another option is to use the formatting utils in https://github.com/celo-org/celo-monorepo/blob/master/packages/mobile/src/utils/formatting.ts#L7

Reply:
+1 for using `<CurrencyDisplay />` here too, we can adapt https://github.com/celo-org/celo-monorepo/blob/6139ec275bc261564e581168b95b112d54b87451/packages/mobile/src/home/CeloDollarsOverview.tsx#L17-L48